src/python/grpcio_tests/tests/unit/_signal_handling_test.py

```diff
@@ -27,7 +27,18 @@
 from tests.unit import test_common
 from tests.unit import _signal_client
 
-_CLIENT_PATH = os.path.abspath(os.path.realpath(_signal_client.__file__))
+_CLIENT_PATH = None
+if sys.executable is not None:
+    _CLIENT_PATH = os.path.abspath(os.path.realpath(_signal_client.__file__))
+else:
+    # Note(rbellevi): For compatibility with internal testing.
+    if len(sys.argv) != 2:
+        raise RuntimeError("Must supply path to executable client.")
+    client_name = sys.argv[1].split("/")[-1]
+    del sys.argv[1]  # For compatibility with test runner.
```

Comment:
Just curious, what will happen if we don't delete the `argv[1]`?

Reply:
Our internal test runner will choke on the argument.